File: src/main/resources/output/IFC4X3_csharp/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProduct/IfcElementType/IfcDistributionElementType/IfcDistributionControlElementType/IfcProtectiveDeviceTrippingUnitType/IfcProtectiveDeviceTrippingUnitType.cs
public class IfcProtectiveDeviceTrippingUnitType : IfcDistributionControlElementType
{
    public IfcProtectiveDeviceTrippingUnitTypeEnum PredefinedType { get; set; }

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : (PredefinedType <> IfcProtectiveDeviceTrippingUnitTypeEnum.USERDEFINED) OR ((PredefinedType = IfcProtectiveDeviceTrippingUnitTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType))
}

File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProduct/IfcElementType/IfcDistributionElementType/IfcDistributionControlElementType/IfcProtectiveDeviceTrippingUnitType/IfcProtectiveDeviceTrippingUnitType.vb
Public Class IfcProtectiveDeviceTrippingUnitType Inherits IfcDistributionControlElementType
    Public PredefinedType As IfcProtectiveDeviceTrippingUnitTypeEnum

    ' === WHERE CLAUSES ===
    ' CorrectPredefinedType : (PredefinedType <> IfcProtectiveDeviceTrippingUnitTypeEnum.USERDEFINED) OR ((PredefinedType = IfcProtectiveDeviceTrippingUnitTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType))
End Class
